src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,158 @@ class DocumentView(BaseLTIView):
+
+    """
+
+    template_name = "core/resource.html"
+
+    def get_context_data(self):
+        """Build context for template rendering of configuration data for the frontend.
+
+        Returns
+        -------
+        dictionary
+            context with configuration data for the frontend
+
+        """
+
+        def _manage_exception(error):
+            logger.warning(str(error))
+            return {"state": "error"}
+
+        lti = LTI(self.request)
+
+        if not lti.is_instructor and not lti.is_admin:
+            raise PermissionDenied
+
+        try:
+            lti.verify()
+        except LTIException as error:
+            app_data = _manage_exception(error)
+        else:
+            app_data = self._get_app_data(lti)
+
+        return {
+            "app_data": json.dumps(app_data),
+            "static_base_url": f"{settings.STATIC_URL}js/",
+            "external_javascript_scripts": settings.EXTERNAL_JAVASCRIPT_SCRIPTS,
+        }
+
+    def _get_app_data(self, lti):
+        """Build app data for the frontend with information retrieved from the LTI launch request.
+
+        Returns
+        -------
+        dictionary
+            Configuration data to bootstrap the frontend:
+            A form is build in react to post data to the LTI Consumer thnough an iframe.
+
+            For instructors only
+            ++++++++++++++++++++
+
+            - lti_select_form_action_url: URL to post data to.
+            - lti_select_form_data: Data to post.
+            - new_document_url: LTI URL to a new document.
+            - new_video_url: LTI URL to a new video.
+            - documents: Documents list with theirs LTI URLs.
+            - videos: Videos list with theirs LTI URLs.
+
+        """
+        documents = DocumentSelectLTISerializer(
+            get_selectable_resources(Document, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        videos = VideoSelectLTISerializer(
+            get_selectable_resources(Video, lti),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        new_uuid = str(uuid.uuid4())
+        app_data = _get_base_app_data()
+        app_data.update(
+            {
+                "frontend": "LTI",
+                "lti_select_form_action_url": reverse("select_lti_view"),
+                "lti_select_form_data": self.request.POST,
+                "new_document_url": self.request.build_absolute_uri(
+                    reverse("document_lti_view", args=[new_uuid])
+                ),
+                "new_video_url": self.request.build_absolute_uri(
+                    reverse("video_lti_view", args=[new_uuid])
+                ),
+                "documents": documents,
+                "videos": videos,
+            }
+        )
+        return app_data
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
```

Comment:
When the playlist does not exist for the context_id posted, we need to create it. 
I did not find a test to prove this scenario.

Reply:
OK, I'll add it.

Reply:
I'm not sure where I should put playlist creation logic.
What are your thoughts about it ?

Reply:
I would say in the "post" view, just after instantiating the "lti" object?

Reply:
OK, thanks